### PR TITLE
Update the linkWithCredential API to call the signUp endpoint

### PR DESCRIPTION
**Pull request by prameshj**
The /signUp API supports linking when Email Enumeration Protection is enabled as well as disabled. /setAccountInfo does not work when Email Enumeration Protection is enabled.

Verified with the test app that anonymous user upgrade works.

### CODE REVIEW

File: packages/auth/src/core/credentials/email.ts [hunk 1]
```diff
@@ -166,7 +169,7 @@ export class EmailAuthCredential extends AuthCredential {
   ): Promise<IdTokenResponse> {
     switch (this.signInMethod) {
       case SignInMethod.EMAIL_PASSWORD:
-        return updateEmailPassword(auth, {
+        return linkEmailPassword(auth, {
```
Comment (prameshj) [inline packages/auth/src/core/credentials/email.ts:172]: Once https://github.com/firebase/firebase-js-sdk/pull/7666 merges, we should also wrap this with reCAPTCHA Enterprise logic.

File: packages/auth/src/api/account_management/email_and_password.ts [hunk 2]
```diff
@@ -69,6 +69,18 @@ export async function updateEmailPassword(
...
 }
 
+// Used for linking an email/password account to an existing idToken. Uses the same request/response
+// format as updateEmailPassword.
+export async function linkEmailPassword(
+  auth: Auth,
+  request: UpdateEmailPasswordRequest
+): Promise<UpdateEmailPasswordResponse> {
+  return _performApiRequest<
+    UpdateEmailPasswordRequest,
```
Comment (renkelvin) [inline packages/auth/src/api/account_management/email_and_password.ts:79]: Sign up request/response?

Comment (prameshj) [reply packages/auth/src/api/account_management/email_and_password.ts]: I was hoping to reuse the Update request/response, but you are right, we need the SignUpRequest and Response to inject recaptcha enterprise fields.

Done.

**Review by renkelvin — APPROVED** [review]

File: packages/auth/src/core/credentials/email.test.ts [hunk 3]
```diff
@@ -237,59 +279,39 @@ describe('core/credentials/email', () => {
...
-            {
-              idToken: 'id-token',
-              refreshToken: 'refresh-token',
-              expiresIn: '1234',
-              localId: serverUser.localId!
-            }
+          // First call without recaptcha token should fail with MISSING_RECAPTCHA_TOKEN error
+          // Though the internal implementation retries with a reCAPTCHA Enterprise token, the second call will fail in this test.
+          // This is because our endpoint mock does not support returning different responses based on different request body params.
+          // TODO(renkelvin) - refactor this once we expose a mockEndpointWithBodyParams or similar method.
```
Comment (renkelvin) [inline packages/auth/src/core/credentials/email.test.ts:285]: What does this method do?

Comment (prameshj) [reply packages/auth/src/core/credentials/email.test.ts:285]: You mean what SHOULD mockEndpointWithBodyParams do? We need a method that will send a different response based on a different request body.

mockEndpointWithParams - is an existing method which sends the specified response, for the request with the given URL Params. But our sign up/sign in are POST methods, so they never match the url param or the mock.

**Review by renkelvin — APPROVED** [review]

**Pull request merged**